lambda/matching.ml

```diff
@@ -1613,7 +1615,7 @@ let split_and_precompile ~arg_id ~arg_lambda pm =
 type cell = {
   pm : initial_clause pattern_matching;
   ctx : Context.t;
-  discr : pattern
+  discr : Pattern_head.t
```

Comment:
This was the last user of `normalize_pat` which is great because pattern heads are a superior version of normalized patterns. I think that `normalize_pat` could be retired starting from this commit.

Reply:
Indeed, I removed it in a small commit.